pagerduty/resource_pagerduty_incident_workflow.go

```diff
@@ -89,35 +200,95 @@ func resourcePagerDutyIncidentWorkflow() *schema.Resource {
 // These inputs get persisted in the state with `generated` set to `true`
 // but then should not be removed by a `terraform apply`
 func customizeIncidentWorkflowDiff() schema.CustomizeDiffFunc {
-	inputCountRegex := regexp.MustCompile(`^(step\.(\d+)\.input)\.#$`)
```

Comment:
I reworked this diff to catch a lot of uncaught diff issues originating from inputs with default values (aka generated) as well as ordering issues. The diff will now use the ordering of inputs from the tfstate, but the values from the config. Numbered comments have been added to make the flow a bit more understandable.

Reply:
Of note is that we now match any input change at any level, not just quantity changes at the top level. This means the "calculate new inputs" logic is now recursive, as it takes in a config path, recurses to the correct level, and then deals with the list of inputs at that level.